### PR TITLE
Archiving GSoD

### PR DESCRIPTION
**Pull request by nate-double-u**
Archiving Google Season of Docs
Pointing people to where they can find CNCF help with GSoD moving forward

contributes to: https://github.com/cncf/mentoring/issues/1027

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,40 +21,38 @@ If you have any questions or need help getting started, please reach out through
...
-| 2017 | GSoC                             | 6           | 8                |
-|      | Outreachy                        | 2           |                  |
-
+| Year | Program                          | Internships  | Total (per year) |
+|------|----------------------------------|--------------|------------------|
+| 2023 | LFX Mentorship                   | 93 (ongoing) | TBD              |
+|      | GSoC                             | 14 (ongoing) |                  |
+| 2022 | LFX Mentorship                   | 87           | 106              |
+|      | GSoC                             | 16           |                  |
+|      | GSoD                             | 3            |                  |
```
Comment (aliok) [inline README.md:40]: We might want to delete GSoD from this table as well.

Not sure if that was intentional (which is also ok) or forgotten

Comment (nate-double-u) [reply README.md:40]: Ya, probably best to remove it.

Comment (nate-double-u) [inline README.md:40]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -21,40 +21,38 @@ If you have any questions or need help getting started, please reach out through
...
+| 2021 | LFX Mentorship                   | 86           | 104              |
+|      | GSoC                             | 16           |                  |
+|      | GSoD                             | 1            |                  |
```
Comment (nate-double-u) [inline README.md:43]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -21,40 +21,38 @@ If you have any questions or need help getting started, please reach out through
...
+|      | Outreachy                        | 1            |                  |
+| 2020 | LFX Mentorship (CommunityBridge) | 50           | 71               |
+|      | GSoC                             | 16           |                  |
+|      | GSoD                             | 4            |                  |
```
Comment (nate-double-u) [inline README.md:47]:
```suggestion
```

File: README.md (hunk 1 above)
Comment (nate-double-u) [reply README.md]: I'm actually reconsidering this a bit... we did run these from within the mentoring program.

Comment (aliok) [reply README.md]: Ok, let's keep it

**Review by aliok — APPROVED** [review]

**Pull request merged**